### ISSUE TITLE
linkml-validate: unexpected inheritance behavior using container pattern

### ISSUE THREAD

**Issue by adhollander** [linkml/linkml#1614]
I am developing a LinkML schema that is borrowing some ideas from Biolink, in particular its association model. I have run into a problem related to inheritance and the container pattern. Illustrating with an example, I have a container class like the following:

```
  Container: 
    class_uri: fsls:Container
    tree_root: true
    slots:
      - nid
      - persons
      - organizations
      - associations
```

The associations slot looks like this:

```
  associations:
    slot_uri: fsls:associations
    multivalued: true
    inlined_as_list: true
    range: Association
```

The Association class looks like

```
  Association:
    class_uri: ppc:Association
    description: >-
      A typed association between two entities, supported by evidence
    slots:
      - name
      - nid
      - subject
      - predicate
      - object
```

and there's a PersonOrganizationAssociation in an is_a relationship with Association which adds startdate, enddate, and title slots and looks like this:

```
  PersonOrganizationAssociation:
    is_a: Association
    class_uri: ppc:PersonOrganizationAssociation
    description: >-
      An association connecting a person with an organization, e.g. "works at"
    slots:
      - startdate
      - enddate
      - title # the name given the role
    defining_slots:
      - subject
      - predicate
      - object
    slot_usage:
      subject:
        range: Person
      object:
        range: Organization
```

If I run linkml-validate over some test data as follows (`linkml-validate -f ttl -C Container  -s PPOD_Core-0.0.4sub8.yaml PPOD_Core_test8.ttl`) I get an error related to the slots that have been added in PersonOrganizationAssociation.

```
 linkml-validate -f ttl -C Container  -s PPOD_Core-0.0.4sub8.yaml PPOD_Core_test8.ttl
Traceback (most recent call last):
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml/validators/jsonschemavalidator.py", line 195, in cli
    data_as_dict = loader.load_as_dict(source=input, **inargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml_runtime/loaders/loader_root.py", line 78, in load_as_dict
    raise NotImplementedError()
NotImplementedError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/bin/linkml-validate", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml/validators/jsonschemavalidator.py", line 197, in cli
    obj = loader.load(source=input, target_class=py_target_class, **inargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml_runtime/loaders/rdflib_loader.py", line 258, in load
    objs = self.from_rdf_graph(g, schemaview=schemaview, target_class=target_class, prefix_map=prefix_map, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml_runtime/loaders/rdflib_loader.py", line 206, in from_rdf_graph
    return [target_class(**x) for x in root_dicts]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml_runtime/loaders/rdflib_loader.py", line 206, in <listcomp>
    return [target_class(**x) for x in root_dicts]
            ^^^^^^^^^^^^^^^^^
  File "<string>", line 7, in __init__
  File "test", line 146, in __post_init__
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml_runtime/utils/yamlutils.py", line 97, in _normalize_inlined_as_list
    self._normalize_inlined(slot_name, slot_type, key_name, keyed, True)
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml_runtime/utils/yamlutils.py", line 182, in _normalize_inlined
    cooked_obj = slot_type(**as_dict(list_entry))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 8, in __init__
  File "test", line 119, in __post_init__
  File "/run/media/adh/Elements/WorkScratch/PPODvenv/lib/python3.11/site-packages/linkml_runtime/utils/yamlutils.py", line 48, in __post_init__
    raise ValueError('\n'.join(messages))
ValueError:  Unknown argument: title = 'Project Scientist'
 Unknown argument: startdate = '2016-01-01'
 Unknown argument: enddate = '2023-01-01'

```

However if I change the range in the associations slot to point directly at the PersonOrganizationAssociation i.e. the slot looking like

```
  associations:
    slot_uri: fsls:associations
    multivalued: true
    inlined_as_list: true
    range: PersonOrganizationAssociation
```

the test data validates without any problem. Something I don't understand is happening with regards to the inheritance relationships but I don't know what it might be. Shouldn't the validator be able to recognize the new slots when PersonOrganizationAssociation subclasses Association?

There is a gist for my LinkML test yaml file PPOD_Core-0.0.4sub8.yaml at https://gist.github.com/adhollander/b7d0db5e00229cd36f517c7cfdf6b57d  and a gist for my test dataset PPOD_Core_test8.ttl at https://gist.github.com/adhollander/823d96ddcd225b9cf969dcf1eacbf7f6.

This is occurring with running linkml version 1.5.7.

**Comment by cmungall** [comment]
Thanks @adhollander 

We are working for a fix for this on Wednesday, see also #1608

**Comment by sierra-moxon** [comment]
I think this has been resolved and can be closed - of course reopen if I made a mistake! :)
